src/components/carousel/index.jsx

```diff
@@ -203,6 +204,17 @@ const Carousel = ({
 
 	const totalSlides = carouselItems.length;
 
+	const emitEvent = (eventName, options) => {
+		EventEmitter.dispatch(eventName, {
+			eventName,
+			ansGalleryId: id,
+			ansGalleryHeadline: label,
+			orderPosition: slide,
+			totalImages: totalSlides,
+			...options,
+		});
+	};
+
```

Comment:
I just want to check this to ensure that we aren't creating a "dead" closure with these events.  Since this is in the render, I think it's potentially creating a new instance of the emitEvent function each time the render is called.  Since `id`, `label`, `slide`, `totalSlides` are all closured into this and this object is used in event handlers, I am curious if these are being properly updated.  (`slide` is a state value, for instance, that could easily change) I just want to test if we need to wrap this up in a useCallback and list those values as dependencies.

Reply:
I was testing this out and saw that there was a small issue with the slide number because it was trying to use it at the same time it was being updated (in `goToSlide()`). I made a small update to fix that.

After testing all the events after that fix, I was seeing `EventEmitter.dispatch()` only being called once each time a triggering event occurred. The data also contained the correct values. I did some more testing with utilizing `useCallback` and saw the same results. If there are "dead" versions of this function being left behind I'm not seeing evidence of them, but it's entirely possible I'm missing something.

Reply:
I may have gone overboard testing it...
https://github.com/WPMedia/arc-themes-components/pull/161
... but I see that change and it looks good now.